### DISCUSSION TITLE
VIDIOC_REQBUFS  only returns 2

### DISCUSSION THREAD

**Discussion by hatemmahmoudi** [umlaeute/v4l2loopback#560] [General]
I am encoutring an issue while attempting to request buffer I only got 2 
```
  struct v4l2_requestbuffers breq = { 0 };
  breq.count = 32;
  breq.type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
  breq.memory = V4L2_MEMORY_MMAP;

 ioctl(mock_v4l2_fd, VIDIOC_REQBUFS, &breq)
```
 
Is a driver limititation ?

**Comment by stephematician** [comment]
The default maximum number of buffers for a device is 2. When the device is created use `max_buffers=32` to allow for up to 32 buffers; e.g.

```
sudo modprobe v4l2loopback max_buffers=32
```
